internal/services/manageddevopspools/managed_devops_pool_resource.go

```diff
@@ -0,0 +1,751 @@
+					if err != nil {
+						return fmt.Errorf("flattening `identity`: %+v", err)
+					}
+					state.Identity = flattenedIdentity
+				}
+
+				if props := model.Properties; props != nil {
+					state.DevCenterProjectResourceId = props.DevCenterProjectResourceId
+					state.MaximumConcurrency = props.MaximumConcurrency
+
+					if agentProfile := props.AgentProfile; agentProfile != nil {
+						if stateful, ok := agentProfile.(pools.Stateful); ok {
+							state.StatefulAgentProfile = flattenStatefulAgentProfileToModel(stateful)
+						} else if stateless, ok := agentProfile.(pools.StatelessAgentProfile); ok {
+							state.StatelessAgentProfile = flattenStatelessAgentProfileToModel(stateless)
+						}
+					}
+
+					if organizationProfile := props.OrganizationProfile; organizationProfile != nil {
+						if azureDevOpsOrganizationProfile, ok := organizationProfile.(pools.AzureDevOpsOrganizationProfile); ok {
+							state.AzureDevOpsOrganizationProfile = flattenAzureDevOpsOrganizationProfileToModel(azureDevOpsOrganizationProfile)
+						}
+					}
+
+					if fabricProfile := props.FabricProfile; fabricProfile != nil {
+						if vmssFabricProfile, ok := fabricProfile.(pools.VMSSFabricProfile); ok {
+							state.VmssFabricProfile = flattenVmssFabricProfileToModel(vmssFabricProfile)
+						}
+					}
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (ManagedDevOpsPoolResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ManagedDevOpsPools.PoolsClient
+
+			id, err := pools.ParsePoolID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+			return nil
+		},
+	}
+}
+
+func (ManagedDevOpsPoolResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return pools.ValidatePoolID
+}
+
+func (ManagedDevOpsPoolResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ManagedDevOpsPoolModel
+			if err := metadata.DecodeDiff(&model); err != nil {
+				return fmt.Errorf("DecodeDiff: %+v", err)
+			}
+
+			// Validate vmss_fabric_profile images with proper handling of computed/unknown values
+			if err := validateVmssFabricProfileImages(metadata, model.VmssFabricProfile); err != nil {
+				return err
+			}
+
+			for _, orgProfile := range model.AzureDevOpsOrganizationProfile {
+				for _, permProfile := range orgProfile.PermissionProfile {
+					if permProfile.Kind != string(pools.AzureDevOpsPermissionTypeSpecificAccounts) {
+						if len(permProfile.AdministratorAccounts) > 0 {
+							return fmt.Errorf("`administrator_account` block is not required when `permission_profile` kind is `%s`", permProfile.Kind)
```

Comment:
Given we're returning an error and preventing this configuration from being provisioned, would it make sense to update the message to something like the below?

imo `is not required` makes it sound optional, but not necessarily incorrect

```suggestion
							return fmt.Errorf("`administrator_account` block must not be specified when `permission_profile` kind is `%s`", permProfile.Kind)
```